common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -85,8 +86,9 @@ public AzureActiveDirectoryWebViewClient(@NonNull final Activity activity,
                                              @NonNull final String redirectUrl) {
         super(activity, completionCallback, pageLoadedCallback);
         mRedirectUrl = redirectUrl;
-        //Creating ClientCertAuthChallengeHandler starts YubiKitManager usb discovery
-        mClientCertAuthChallengeHandler = new ClientCertAuthChallengeHandler(getActivity());
+        //Creating ClientCertAuthChallengeHandler starts smartcard usb discovery
+        mClientCertAuthChallengeHandler = new ClientCertAuthChallengeHandler(getActivity(),
```

Comment:
Calling a constructor in a constructor is _usually_ a sign your class needs to be accepting its dependency, rather than creating it. Consider passing in the instance of `ClientCertAuthChallengeHandler` or supplying an interface for it. This can make these classes easer to test separately from each other

Reply:
https://java-design-patterns.com/patterns/dependency-injection/